core/tests/ledger_cleanup.rs

```diff
@@ -377,23 +396,17 @@ mod tests {
                 time = Instant::now();
             }
 
-            if shreds.lock().unwrap().len() < 50 {
+            if shreds.lock().unwrap().len() < 50 && !pre_generate_data {
```

Comment:
Can you swap the order of statements so we short-circuit when `pre_generate_data == true` and avoid grabbing the lock?